catalog/ui/catalog-ui-oauth/src/main/java/org/codice/ddf/catalog/ui/oauth/app/OAuthApplication.java

```diff
@@ -201,7 +202,12 @@ String processCodeFlow(Response res, String code, Map<String, Object> state) {
 
     JWT idToken = oidcTokens.getIDToken();
     AccessToken accessToken = oidcTokens.getAccessToken();
-    RefreshToken refreshToken = oidcTokens.getRefreshToken();
+    RefreshToken refreshToken = null;
+
+    if (!userId.startsWith(GUEST)) {
```

Comment:
✏️  I approved already, but it might be better to actually use SubjectUtils.isGuest(...) instead.

Reply:
The problem is the subject is not attached to the thread for some reason at this point